src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineCloud.java

```diff
@@ -162,21 +173,20 @@ public void addConfiguration(InstanceConfiguration config) {
                     public Node call() throws Exception {
                         long startTime = System.currentTimeMillis();
                         LOGGER.log(Level.INFO, String.format("Waiting %dms for node %s to connect", config.getLaunchTimeoutMillis(), node.getNodeName()));
-                        while ((System.currentTimeMillis() - startTime) < config.getLaunchTimeoutMillis()) {
-                            LOGGER.log(Level.INFO, String.format("%dms elapsed waiting for node %s to connect", System.currentTimeMillis() - startTime, node.getNodeName()));
-                            Computer c;
-                            try {
-                                c = node.toComputer();
-                                if (c != null) {
-                                    c.connect(false).get();
-                                }
-                            } catch (Exception e) {
-                                LOGGER.log(Level.WARNING, String.format("Exception waiting for node %s to connect", node.getNodeName()), e);
+                        try {
+                            Computer c = node.toComputer();
+                            if (c != null) {
+                                c.connect(false).get(config.getLaunchTimeoutMillis(), TimeUnit.MILLISECONDS);
```

Comment:
If using the Future's get() method with a timeout argument, then it probably makes sense to try/catch the TimeoutException, and have the error case live in the catch block.

Reply:
Its cached as Exception below..
Im not sure if we want extra catch for it..
It will be maybe different log message nothing more..

Reply:
I'm tend to be wary of catch blocks with the type "Exception", as they can sometime cover up problems unintentionally. I suggest having separate catch blocks for each type which can be thrown, with meaningful log messages and error handling for each.

Reply:
I cannot catch it as `TimeoutException | Exception e` and Im really not sure if we want to duplicate this code.. Just error message maybe will be different.. even not because we are re throwing exception into log..

Reply:
To clarify, I mean removing the "Exception" type from the catch block and replacing it with the types which actually get thrown.

Reply:
Fixed